components/net_traits/lib.rs

```diff
@@ -525,6 +525,8 @@ pub enum ConstellationMsg {
 /// Network errors that have to be exported out of the loaders
 #[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize, HeapSizeOf)]
 pub enum NetworkError {
+    /// Request attempted on bad port - https://fetch.spec.whatwg.org/#block-bad-port
+    BadPort,
```

Comment:
I primarily did this just so that I didn't have to check the string value returned in the test for this error case.  

In the test, is it okay to just check the string value of the `Internal` error to confirm that the correct error was returned?